Documentation/metrics.md

```diff
@@ -1,29 +1,48 @@
-<!-- AUTOGENERATED BY glean_parser.  DO NOT EDIT. -->
```

Comment:
A `metrics.md` file is not required anymore.
Public Mozilla projects can rely on the Glean Dictionary for documentation, e.g. Focus iOS is here: https://dictionary.telemetry.mozilla.org/apps/focus_ios

We recommend replacing the generated file with some content pointing there, e.g. see [Fenix' metric.md](https://github.com/mozilla-mobile/fenix/blob/main/docs/metrics.md).
This is not directly related to changes in this PR, so could also be done separately (or the file could be kept, that's up to the project)